software/base/src/main/java/brooklyn/entity/basic/AbstractSoftwareProcessDriver.java

```diff
@@ -251,6 +264,183 @@ public InputStream getResource(String url) {
         return resource.getResourceFromUrl(url);
     }
 
+    /**
+     * Files and templates to be copied to the server <em>before</em> pre-install. This allows the {@link #preInstall()}
+     * process to have access to all required resources.
+     * <p>
+     * Will be prefixed with the entity's {@link #getInstallDir() install directory} if relative.
+     *
+     * @see SoftwareProcess#PRE_INSTALL_FILES
+     * @see SoftwareProcess#PRE_INSTALL_TEMPLATES
+     * @see #copyRuntimeResources()
+     */
+    public void copyPreInstallResources() {
```

Comment:
I presume these were copied unmodified from AbstractSoftwareProcessSshDriver? Therefore not reviewing thoroughly. If anything (beyond calling abstract methods at the appropriate points) has changed, then let me know and I'll review more thoroughly.